### DISCUSSION TITLE
No Component for Form Level Errors?

### DISCUSSION THREAD

**Discussion by arkmech** [svecosystem/formsnap#158] [General]
Title^

**Comment by huntabyte** [comment]
Since there isn't a top-level `Form` component, there is nothing the 	`<FormErrors />` would receive context from, meaning you'd have to pass the errors into the component.

I'll look into seeing if this makes sense. Need to confirm the best accessibility practices for handling more general "form-level" errors. If there is a case for it, I'll consider adding a component for it.

**Reply by Mardoxx** [reply]
I’d really appreciate this feature as well. Right now, I’m handling form-level errors by passing a custom error prop, but having a more integrated solution would make form error handling much cleaner and more consistent. A holistic approach to managing both field-level and form-wide errors would be a great addition.

**Reply by xyangst** [reply]
for now you could just use superform messages i think
https://superforms.rocks/concepts/messages

**Reply by filiabel** [reply]
I know you're busy with packing up the new `shadcn-svelte` release - and thanks for all the immense work you're doing 👏 Chiming in here as I came across the discussion just now.

I think a separate component like `<Form.Errors {form} />` (as an example) would be very useful, tapping into [`$errors._errors`](https://superforms.rocks/concepts/error-handling#form-level-and-array-errors) making Formsnap encapsulate all Superform errors. 
E.g. when having wrong credentials (username and/or password), rate limiting, or other errors not directly related to a form field.

You could hack it like code below, but that causes a TS error on `name` prop. 
```svelte
<Form.Field {form} name="_errors">
	<Form.FieldErrors />
</Form.Field>
```
